File: frontend/src/app/shared/service/admin/freetrial/FreeTrialformuleAdmin.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';

import {environment} from '../../../../../environments/environment';

import {FreeTrialformuleDto} from '../../../model/freetrial/FreeTrialformule.model';
import {FreeTrialformuleCriteria} from '../../../criteria/freetrial/FreeTrialformuleCriteria.model';
import {AbstractService} from "../../../../zynerator/service/AbstractService";


@Injectable({
    providedIn: 'root'
})
export class FreeTrialformuleAdminService extends AbstractService<FreeTrialformuleDto, FreeTrialformuleCriteria> {
    constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    get API() {
        return environment.apiUrlAlcservice + 'admin/freeTrialformule/';
    }

    public constrcutDto(): FreeTrialformuleDto {
        return new FreeTrialformuleDto();
    }

    public constrcutCriteria(): FreeTrialformuleCriteria {
        return new FreeTrialformuleCriteria();
    }
}

File: backend-ms1/src/main/java/ma/zs/zyn/service/facade/admin/freetrial/FreeTrialformuleAdminService.java
package ma.zs.zyn.service.facade.admin.freetrial;

import java.util.List;
import ma.zs.zyn.bean.core.freetrial.FreeTrialformule;
import ma.zs.zyn.dao.criteria.core.freetrial.FreeTrialformuleCriteria;
import ma.zs.zyn.zynerator.service.IService;



public interface FreeTrialformuleAdminService extends  IService<FreeTrialformule,FreeTrialformuleCriteria>  {

    List<FreeTrialformule> findByInscriptionId(Long id);
    int deleteByInscriptionId(Long id);
    long countByInscriptionId(Long id);




}
